### ISSUE TITLE
Fixing files using the dbt_utils.surrogate_key function results in compiled code being inserted into model definitions

### ISSUE THREAD

**Issue by cwcartmell** [sqlfluff/sqlfluff#2249]
I noticed this issue while formatting a much more complex query than the one presented here. In different queries that I managed to get to display the same problem the offending line was not necessarily always directly after the jinja that it's compiled from. It also doesn't appear to matter what error is discovered and fixed. I used an indentation error in this example, but other ones seem to do the trick too.

## Expected Behaviour

This dbt model definition file (surrogate_key_error.sql):

```sql
SELECT
    'value' AS variable,
    {{ dbt_utils.surrogate_key(["variable"]) }} AS hashed_column,
     'value' AS variable_2
```

Should be modified by `sqlfluff fix -f --dialect=redshift ./models/surrogate_key_error.sql` by removing a single space from the start of line 4, like so:

```sql
SELECT
    'value' AS variable,
    {{ dbt_utils.surrogate_key(["variable"]) }} AS hashed_column,
    'value' AS variable_2
```

## Observed Behaviour

```bash
$ sqlfluff fix -f --dialect=redshift -v ./models/surrogate_key_error.sql
==== sqlfluff ====
sqlfluff:                0.7.1 python:                  3.8.8
implementation:        cpython dialect:              redshift
verbosity:                   1 templater:                 dbt
dbt:                   =0.20.2

==== finding fixable violations ====
=== [ path: ./models/surrogate_key_error.sql ] ===

=== [dbt templater] Sorting Nodes...
=== [dbt templater] Compiling dbt project...
=== [dbt templater] Project Compiled.
== [models/surrogate_key_error.sql] FAIL
L:   4 | P:   6 | L003 | Indentation not consistent with line #3
==== fixing violations ====
1 fixable linting violations found
FORCE MODE: Attempting fixes...
Persisting Changes...
== [models/surrogate_key_error.sql] PASS
Done. Please check your files to confirm.
```

Results in:

```sql
SELECT
    'value' AS variable,
    {{ dbt_utils.surrogate_key(["variable"]) }} AS hashed_column,
    md5(cast(coalesce(cast(variable AS varchar), '') AS varchar)) AS hashed_column
```

## Dialect

redshift

## Version

```bash
$ sqlfluff --version
sqlfluff, version 0.7.1
$ python --version
Python 3.8.8
$ dbt --version
installed version: 0.20.2
   latest version: 1.0.0

Your version of dbt is out of date! You can find instructions for upgrading here:
https://docs.getdbt.com/docs/installation

Plugins:
  - bigquery: 0.20.2
  - snowflake: 0.20.2
  - redshift: 0.20.2
  - postgres: 0.20.2
```

## Configuration

```toml
[tool.sqlfluff.core]
templater = "dbt"

[tool.sqlfluff.templater.dbt]
project_dir = "."
```

**Comment by jpy-git** [comment]
Thanks for raising this @cwcartmell! 

@barrywhart has being doing a lot of work to fix similar issues since the last release so I wonder if this has already been fixed? Looks similar to #1952 and #2056. Maybe another test case to look at 😄

**Comment by barrywhart** [comment]
Yes, this should be fixed in the next release. If you want to confirm before the release, you can test with the latest code in `main`.

**Comment by barrywhart** [comment]
I'm going to go ahead and close this. Please let us know if the next release doesn't fix it. (I think it'll be coming out in the next week or so.)
